internal/state/graph/httproute.go

```diff
@@ -281,57 +281,100 @@ func bindRouteToListeners(r *Route, gw *Gateway) {
 			continue
 		}
 
-		// Case 3 - winning Gateway
+		// Case 3: Attachment is not possible because Gateway is invalid
```

Comment:
After adding this new case, the linter complained about this function being too complex, so I refactored it to reduce the complexity. 

cc @sjberman @pleshakov